File: src/components/info/weapon/elemental.ts
import Weapon, { WeaponGroup } from "./_base";
import { EFFECTS as eff } from "../_effect";

let weapon: Weapon[] = [];

weapon.push({
    name: "elemental",
    effects: [{ effect: eff.FLOOR_POTENCY, amount: 1.7 }],
});

const ELEMENTAL: WeaponGroup = {
    name: "elemental",
    weapons: weapon,
};

export default ELEMENTAL;

File: _data/weapons/elemental.py
from typing import List

from ._weapon import Weapon
from ._weapon_groups import WeaponGroup
from effect import EffectTypes as ET, Effect

TYPE = WeaponGroup.ELEMENTRAL

weapons: List[Weapon] = []

weapons.append(
    Weapon(
        "elemental weapon",
        (Effect(ET.FLOOR_POT, 1.7),),
        TYPE,
    )
)
